infrastructure/src/main/java/org/corfudb/infrastructure/BatchWriterOperation.java

```diff
@@ -20,7 +20,10 @@
         PREFIX_TRIM,
         SEAL,
         RESET,
-        TAILS_QUERY
+        LOG_TAIL_QUERY,
+        TAILS_QUERY,
+        STREAMS_ADDRESS_QUERY,
+        LOG_ADDRESS_SPACE_QUERY
```

Comment:
I don't think we should serialize readers.

Reply:
As per the comment below we agreed to leave it serialized as it is not thread safe and also the fact that this is only used on reconfiguration does also not make it critical to go through the batchProcessor.